R/npm-assess.R

```diff
@@ -0,0 +1,117 @@
+
+
+#' NPM assess function
+#' 
+#' This function aims to apply the various scoring functions 
+#' across a data.frame on a row by row basis.
+#' It is expected that you would use this function in conjunction with a 
+#' call to `lapply` or on a single row.
+#' The function returns a data.frame containing the A score, C score, 
+#' NPM score and NPM assessment.
+#' 
+#' @export
+#' @param row, the row from an NPMScore call
+#' @return a data.frame containing A score, C score, NPM score, and NPM assessment for each row
+NPMAssess <- function(row) {
+
+    stopifnot("Score columns not detected. Please ensure you have called NPMScore on your data" = 
+        c("energy_score", "sugar_score", "fat_score", 
+        "protein_score", "fvn_score", "fibre_score") %in% names(row)
+        
+    )
+
+    A_score <- A_scorer(energy_score = row[['energy_score']],
+                        sugar_score = row[['sugar_score']], 
+                        fat_score = row[['fat_score']], 
+                        sodium_score = row[['sodium_score']])
+
+    C_score <- C_scorer(fvn_score = row[['fvn_score']],
+                        protein_score = row[['protein_score']], 
+                        fibre_score = row[['fibre_score']])   
+
+    NPM_score <- NPM_total(A_score, C_score, row[['fvn_score']], row[['fibre_score']])
+
+    NPM_assessment <- NPM_assess(NPM_score, row[['product_type']])
+
+    npm_df <- data.frame("A_score" = A_score, "C_score" = C_score, "NPM_score" = NPM_score, "NPM_assessment" = NPM_assessment)
```

Comment:
Create data.frame here to get expected output when calling `lapply` on this function. This also makes it easier to preserve types.